src/prefect/utilities/executors.py

```diff
@@ -174,11 +174,42 @@ def multiprocessing_safe_run_and_retrieve(
         with prefect.context(context):
             logger.debug(f"{name}: Executing...")
             return_val = fn(*args, **kwargs)
+            logger.debug(f"{name}: Execution successful.")
     except Exception as exc:
         return_val = exc
+        logger.debug(
+            f"{name}: Encountered a {type(exc).__name__}, "
+            f"returning details as a result..."
+        )
+
+    try:
+        pickled_val = cloudpickle.dumps(return_val)
+    except Exception as exc:
+        base_msg = (
+            f"Failed to pickle result of type {type(return_val).__name__!r} with "
+            f'exception: "{type(exc).__name__}: {str(exc)}".'
+        )
+        logger.error(f"{name}: {base_msg}")
```

Comment:
```suggestion
        logger.error(f"{name}: {base_msg}", exc_info=True)
```

Reply:
I'm not sure it makes sense to log the traceback here. It's just going to point them into this bit of code and a `Pickle.dumps` call. The returned error will include a decent traceback that should be sufficient and I worry printing several tracebacks will just be confusing.

Reply:
(note I've already put the exception type and message into the `base_msg` so it's mostly redundant other than the traceback)